Add missions to prevent the player from becoming stranded in Coalition space (#5178)
endless-sky/endless-sky: data/coalition/coalition missions.txt

mission "lost in coalition"
	landing
	invisible
	source
		government "Coalition" "Heliarch"
	to offer
		has "Coalition: Contributor: done"
	on offer
		require "Jump Drive" 0
		event "lost in coalition" 14
		fail

event "lost in coalition"

mission "Coalition: Lost: Marker"
	name "Foreign Aid"
	description "The Heliarchs have noticed that you've lost your jump drive, being now stranded in Coalition space, and have offered to help. Head to <destination> to speak with them."
	landing
	source
		near "Ekuarik" 1 100
	destination "Ring of Wisdom"
	to offer
		has "event: lost in coalition"
	on offer
		conversation

> 			`Another landing, another day spent stranded in this alien region, with no other humans in sight. Be it with the help of a Heliarch agent, via their own translation devices, or just by gesturing and hoping for the best, you've managed to get by just fine, but the thought that you may live the rest of your life living among an alien civilization refuses to leave your mind.`

Reviewer: Given that the time that I lost my JD in Coalition space I hadn't even realized it until much later, I'm not so sure if this is how we should start this conversation. The 14 day wait in Remnant space given how small it is means that the player is more likely to realize their mistake, but in Coalition space it could take longer than 14 days for the player to realize.

I'd suggest getting rid of this first line and instead starting with the player being approached by the Heliarchs, who comment on the player's lack of a JD. Then, have a choice where the player can either thank the Heliarchs for their generosity, or be shocked that they've lost their JD.

			`	Trying to take your mind off of it all, you start walking around <origin>, looking for some place to eat, but are stopped by a Heliarch delegation shortly after you leave your ship.`
			`	"Cautiously observed during the past few weeks, scans of your ship have been, Captain <last>," the Saryd begins.`
			`	"Unlike previous scans, a jump drive in your ship, our scanners found not," the Kimek continues.`

> 			`	"Though appreciate the work you here perform, we do, for you to be cut-off from the galaxy, unlike the Quarg we wish not," the Arach says. "Discussed much, our elders have, and decided to help you, they have. To <destination> you must go, there discussed, the terms will be."`

Reviewer: Repeat "have."
```suggestion
			`	"Though appreciate the work you here perform, we do, for you to be cut-off from the galaxy, unlike the Quarg we wish not," the Arach says. "Discussed much, our elders have, and to help you, they decided. To <destination> you must go, there discussed, the terms will be."`
```

> 			`	You thank them for bringing you this information, and go back to your ship to prepare a route to <destination>.`

Reviewer: This is where you'd put the choice. Also, you've forgotten an endpoint.

	to complete

> 		has "Coalition: Lost: Offer: active"

Reviewer: I'd instead change this to never complete, then have accepting the next mission fail this one.
```suggestion
		never
```

mission "Coalition: Lost: Offer"
	name "Jump Drive Loan"

> 	description "The Heliarchs have provided you a jump drive to leave Coalition space. Bring another jump drive to <destination> by <date>, to replace the one you borrowed, or they will be extremely angry by you breaking their trust."

Reviewer: ```suggestion
	description "The Heliarchs have provided you a jump drive to leave Coalition space. Bring another jump drive to <destination> by <date>, to replace the one you borrowed, or they will be extremely angered by you breaking their trust."
```

	deadline 333
	landing
	source "Ring of Wisdom"
	to offer
		has "Coalition: Lost: Marker: active"
	on offer
		conversation
			`The Heliarchs are waiting for you when you land. You see no civilians as you look around, but there are enough armed guards to man several of their Punishers. Said Punishers are all parked near your own ship.`
			`	The Saryd starts speaking, "As to be trusted, you have been deemed, provide you with a jump drive, so that stranded you are not, we will."`
			`	"Mind you, not a gift this is, but a loan, Captain," the Kimek continues. "Another one to replace this one, you must bring us, so that losing a copy of such valuable equipment, we are not."`
			`	"By <date>, brought here the replacement jump drive must be. If any longer you take, assume that betrayed our trust you have, we will, and an enemy, considered you will be," the Arach finishes explaining. "Agree with those terms, do you, Captain?"`
			choice

> 				`	"Yes, I promise I'll be back here with a second jump drive by <date>."`

Reviewer: It seems odd to me for the player to repeat the date, especially given that it is reiterated in the next line, so I'd remove that part.
```suggestion
				`	"Yes, I promise I'll return with a second jump drive."`
```

				`	"Sorry, I need more time to think on this."`
					defer

> 			`	Their engineers proceed to install the jump drive in your ship. A couple of the guards come a bit closer during the process.`
> 			`	When it's all done, you're reminded that you need to bring another jump drive here by <date>.`

Reviewer: ```suggestion
			`	Their engineers proceed to install the jump drive in your ship. A couple of the guards come a bit closer during the process. When it's all done, you're reminded that you need to bring another jump drive here by <date>.`
```

				accept

> 	on accept
> 		outfit "Jump Drive" 1

Reviewer: ```suggestion
	on accept
		outfit "Jump Drive" 1
		fail "Coalition: Lost: Marker"
```
